cli/src/account_history.rs

```diff
@@ -27,17 +27,27 @@ impl<T: ToString> History<T> for AccountHistory {
     fn write(&mut self, val: &T) {
         let entry = val.to_string();
 
-        if let Some(command) = self.history.front() {
-            // If the last used command is the same, dont add it
-            if command == &entry {
-                return;
-            }
+        // If the last used command is the same, dont change anything
+        if matches!(self.history.front(), Some(command) if command == &entry) {
+            return;
         }
 
-        if self.history.len() == self.max {
-            self.history.pop_back();
+        // We only need to check size if we dont remove an item
+        match self.history.binary_search(&entry) {
```

Comment:
The history is not ordered, so this not safe

Reply:
ah i misread, thought it meant maintain ordered